packages/sales-api-service/src/services/transactions/__tests__/process-transaction-queue.spec.js

```diff
@@ -410,6 +418,57 @@ describe('transaction service', () => {
         await processQueue({ id: finalisedTransaction.id })
         expect(processRecurringPayment).toHaveBeenCalledWith(rprSymbol, expect.any(Contact))
       })
+
+      it('checks for existing recurring payments with the same agreementId', async () => {
+        const agreementId = Symbol('agreementId')
+        const mockRecurringPayment = { agreementId }
+        const finalisedTransaction = mockFinalisedTransactionRecord()
+        processRecurringPayment.mockResolvedValueOnce({ recurringPayment: mockRecurringPayment })
+        AwsMock.DynamoDB.DocumentClient.__setResponse('get', { Item: finalisedTransaction })
+
+        await processQueue({ id: finalisedTransaction.id })
+
+        expect(findNewestExistingRecurringPaymentInCrm).toHaveBeenCalledWith(agreementId)
+      })
+
+      it("assigns the new RecurringPayment as the existing RecurringPayment's nextRecurringPayment", async () => {
+        const agreementId = Symbol('agreementId')
+        const mockRecurringPayment = { agreementId }
+        const mockExistingRecurringPayment = { bindToEntity: jest.fn() }
+        const finalisedTransaction = mockFinalisedTransactionRecord()
+        processRecurringPayment.mockResolvedValueOnce({ recurringPayment: mockRecurringPayment })
+        AwsMock.DynamoDB.DocumentClient.__setResponse('get', { Item: finalisedTransaction })
+        findNewestExistingRecurringPaymentInCrm.mockReturnValueOnce(mockExistingRecurringPayment)
+
+        await processQueue({ id: finalisedTransaction.id })
+
+        expect(mockExistingRecurringPayment.bindToEntity).toHaveBeenCalledWith(
+          RecurringPayment.definition.relationships.nextRecurringPayment,
+          mockRecurringPayment
+        )
+      })
+
+      it('does not attempt to update multiple recurring payment entities if no pre-existing recurring payment is found', async () => {
+        const agreementId = Symbol('agreementId')
+        const mockRecurringPayment = { agreementId }
+        const finalisedTransaction = mockFinalisedTransactionRecord()
+        processRecurringPayment.mockResolvedValueOnce({ recurringPayment: mockRecurringPayment })
+        AwsMock.DynamoDB.DocumentClient.__setResponse('get', { Item: finalisedTransaction })
+        findNewestExistingRecurringPaymentInCrm.mockReturnValueOnce(false)
+
+        await processQueue({ id: finalisedTransaction.id })
+
+        expect(persist.mock.calls[0][0]).toEqual([
```

Comment:
don't think should be so many expect statements, maybe could look into using snapshot